sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -584,6 +584,8 @@ def build_component_spec_for_group(
                 input_name].artifact_type.CopyFrom(
                     type_utils.bundled_artifact_to_artifact_proto(
                         channel.channel_type))
+            component_spec.input_definitions.artifacts[
+                input_name].is_artifact_list = channel.is_artifact_list
```

Comment:
Write the `isArtifactList` field when surfacing an artifact list for a sub-DAG when surfacing an inner artifact list